src/main/java/com/oath/halodb/HaloDBInternal.java

```diff
@@ -323,34 +423,45 @@ private InMemoryIndexMetaData writeRecordToFile(Record record) throws IOExceptio
         return currentWriteFile.writeRecord(record);
     }
 
-    private void rollOverCurrentWriteFile(Record record) throws IOException, HaloDBException {
+    private void rollOverCurrentWriteFile(Record record) throws IOException {
         int size = record.getKey().length + record.getValue().length + Record.Header.HEADER_SIZE;
+        if ((currentWriteFile == null || currentWriteFile.getWriteOffset() + size > options.getMaxFileSize())
+            && !isClosing) {
```

Comment:
Why rollover not allowed during closing?  Looks like this is not relevant to snapshot feature. If required, please open  a new PR to fix that.

Reply:
I just refactored the structure of the old function. There's no logic change in this.